conda/core/package_cache_data.py

```diff
@@ -782,143 +785,156 @@ def execute(self):
 
         assert not context.dry_run
 
-        if not self.paired_actions:
-            return
-
-        if not context.verbose and not context.quiet and not context.json:
-            print(
-                "\nDownloading and Extracting Packages:",
-                end="\n" if IS_INTERACTIVE else " ...working...",
-            )
-        else:
-            log.debug(
-                "prepared package cache actions:\n"
-                "  cache_actions:\n"
-                "    %s\n"
-                "  extract_actions:\n"
-                "    %s\n",
-                "\n    ".join(str(ca) for ca in self.cache_actions),
-                "\n    ".join(str(ea) for ea in self.extract_actions),
-            )
+        with get_progress_bar_context_manager() as pbar_context:
+            if self._executed:
+                return
+            if not self._prepared:
+                self.prepare()
 
-        exceptions = []
-        progress_bars = {}
-        futures: list[Future] = []
-
-        cancelled_flag = False
-
-        def cancelled():
-            """
-            Used to cancel download threads.
-            """
-            nonlocal cancelled_flag
-            return cancelled_flag
-
-        with signal_handler(conda_signal_handler), time_recorder(
-            "fetch_extract_execute"
-        ), ThreadPoolExecutor(
-            context.fetch_threads
-        ) as fetch_executor, ThreadPoolExecutor(EXTRACT_THREADS) as extract_executor:
-            for prec_or_spec, (
-                cache_action,
-                extract_action,
-            ) in self.paired_actions.items():
-                if cache_action is None and extract_action is None:
-                    # Not sure when this is reached.
-                    continue
+            if not context.verbose and not context.quiet and not context.json:
+                print(
+                    "\nDownloading and Extracting Packages:",
+                    end="\n" if IS_INTERACTIVE else " ...working...",
+                )
+            else:
```

Comment:
Shouldn't we log regardless the stdout config?

Reply:
I didn't change this line. I just moved it from one place to another.

Let's not worry about it for this pull request.